Assets/Scripts/Entities/Containers/ItemMap.cs

```diff
@@ -0,0 +1,96 @@
+﻿namespace Assets.Scripts.Entities.Items
+{
+    using System.Collections.Generic;
+    using System.IO;
+    using log4net;
+    using SimpleJSON;
+    
+    /// <summary>
+    /// Container class for holding static references to game items.
+    /// </summary>
+    class ItemMap
+    {
+        private static readonly ILog log = LogManager.GetLogger(typeof(ItemMap));
+
+        public Dictionary<int, Item> items;
+
+        /// <summary>
+        /// Singleton accessor
+        /// </summary>
+        public static ItemMap INSTANCE
+        {
+            get
+            {
+                if (_instance == null)
+                {
+                    _instance = new ItemMap();
+                }
+
+                return _instance;
+            }
+        }
+        /// <summary>
+        /// Instance holder.
+        /// </summary>
+        private static ItemMap _instance;
+
+        public Item this[int itemId]
+        {
+            get
+            {
+                return items[itemId];
+
+            }
+        }
+
+        public ItemMap()
+        {
+            items = new Dictionary<int, Item>();
+            loadItems();
+        }
+
+        /// <summary>
+        /// Loads item data from disk.
+        /// </summary>
+        public void loadItems()
+        {
+            JSONNode json = null;
+
+            try
+            {
+                string rawItemData = File.ReadAllText(Item.ITEM_PATH);
+                json = JSON.Parse(rawItemData)["items"];
```

Comment:
could you reformat the item.json to not be {items:[{},{}]} and just have it as an array like: [{},{}]?
the items part of the object is implied by the filename

Reply:
That is a valid point, I suppose since being a first iteration on the item parsing/building I wasn't super focused on whether to have an implicit root level collection, or to have everything nested in a top-level "items" node.

In the future I foresee the need to potentially split out different types of item collections, where each sub-list consists of similar themed' of items.
e.g.

`{
    "wizard_items": [
        {
            "id": 1,
            "name":"Staff of Light"
        },
        {
            "id": 2,
            "name":"Inniates Robe"
        }
    ],
    "potions": [
        {
            "id": 1,
            "name":"Minor Health Potion"
        },
        {
            "id": 4,
            "name":"Wizards Mind Bomb"
        }
    ],
    "quest_items": [
        {
            "id": 5,
            "name":"Ancient Medalion"
        }
    ]
}`


When item progression, usage and categorisation becomes clearer, this is definately something that could be refactored/re-addressed.  I think the idea of having an item model that allows named sub-lists will be more manageable in the future as opposed to referencing certain item categories by an arbtirary index. 

Fair point though, in this current iteration, this does add a tiny amont of more processing, I will need to revist this,